functional/FunctionalTest.java

```diff
@@ -2856,19 +2861,43 @@ public void run() {
       // Upload an object until we detect an event, this is done
       // in a loop because bucket listening is called in a thread
       // and may or may not be executed at this point.
-      while (bl.eventsReceived < 1) {
+      for (int i = 0; i < 20; i++) {
         client.putObject(bucketName, "prefix-random-suffix", file1kb);
-        Thread.sleep(300);
+        Thread.sleep(500);
+        if (bl.eventsReceived > 0 || bl.lastException != null) {
+          break;
+        }
       }
 
+      // Fail for any exception but ignore NotImplemented error
```

Comment:
Maybe the cleanup code of removing the bucket and object should be moved to this part before you start checking for errors.

Reply:
> Maybe the cleanup code of removing the bucket and object should be moved here before you start checking for errors.

Nice catch! fixed